app/models/foreman_tasks/concerns/host_action_subject.rb

```diff
@@ -15,26 +15,30 @@ def available_locks
       module ClassMethods
         # TODO: This should get into the Foreman core, extracting the
         # +importHostAndFacts+ method into two
-        def importHost(hostname, certname, proxy_id = nil)
+        def import_host(hostname, certname, facts, proxy_id = nil)
+          raise(::Foreman::Exception.new("Invalid Facts, must be a Hash")) unless facts.is_a?(Hash)
           raise(::Foreman::Exception.new("Invalid Hostname, must be a String")) unless hostname.is_a?(String)
 
           # downcase everything
           hostname.try(:downcase!)
           certname.try(:downcase!)
+          facts['domain'].try(:downcase!)
```

Comment:
It's very not obvious, because I guess the facts are passed around by reference and you're modifying it here so it's downcased later?

Maybe create a normalize_facts method and call it from `Plan`? At least it'd be clearer what's going on imho.